edx_arch_experiments/summaryhook_aside/block.py

```diff
@@ -0,0 +1,80 @@
+"""Xblock aside enabling OpenAI driven summariess"""
+
+from django.conf import settings
+from django.template import Context, Template
+from web_fragments.fragment import Fragment
+from xblock.core import XBlockAside
+
+from edx_arch_experiments.summaryhook_aside.waffle import summary_enabled
+
+summary_fragment = """
+<div class="summary-hook">
+  <div id="ai-spot-root"
+    data-url-api="{{data_url_api}}"
+    data-text-identifier="{{data_text_identifier}}"
+  ></div>
+  <script type="text/javascript" src="{{js_url}}" defer="defer"></script>
+</div>
+"""
+
+
+def _render_summary(context):
+    template = Template(summary_fragment)
+    return template.render(Context(context))
+
+
+def _children_have_summarizable_content(block):
+    """
+    Only if a unit contains HTML blocks with sufficient text in them
+    is it worth injecting the summarizer.
+    """
+    children = block.get_children()
+    for child in children:
+        if (
+                getattr(child, 'category', None) == 'html'
+                and hasattr(child, 'get_html')
+                and len(child.get_html()) > settings.SUMMARY_HOOK_MIN_SIZE
+        ):
+            return True
+
+    return False
+
+
+class SummaryHookAside(XBlockAside):
+    """
+    XBlock aside that injects AI summary javascript
+    """
+
+    @XBlockAside.aside_for('student_view')
+    def student_view_aside(self, block, context=None):  # pylint: disable=unused-argument
+        """
+        Renders the aside contents for the student view
+        """
+        fragment = Fragment('')
+
+        if not _children_have_summarizable_content(block):
+            return fragment
+
+        fragment.add_content(
+            _render_summary(
+                {
+                    'data_url_api': settings.SUMMARY_HOOK_HOST,
+                    'data_text_identifier': '.xmodule_HtmlBlock',
```

Comment:
Please don't use the term 'xmodule' (I know it's in the package, but we're trying to kill references to that name everywhere).

Reply:
this is actually one of the classes of the div in the output, it's here so the summary javascript knows what to scoop out. I'll have a look to see if there is another class that I could use instead.

Reply:
looks like I could use xblock-student_view-html so I'll try that locally and update

Reply:
Ah, if it's mapping to an existing page identifier, don't worry about it too much. We'll probably never be able to change those for backwards compatibility reasons anyhow...